Add note on officially supported platforms (#22344)
grpc/grpc: src/cpp/README.md

In the C++ world, there's no universally accepted standard for managing project dependencies.
Therefore, gRPC supports several major build systems, which should satisfy most users.

## Supported Platforms

* Supported: These platforms are officially supported. We test our code on these
platform and have automated continuous integration tests for them.

> * Best Effort: We may or may not have continous integration tests for these, but we are fairly confident that gRPC C++ would work on them.

Reviewer: Do we have platforms for `Best Effort`?

Author: just added arm, ios and android, but I do not have version information for them. I guess we can fill it in as we get more information

> * Best Effort: We may or may not have continous integration tests for these, but we are fairly confident that gRPC C++ would work on them.

Reviewer 2: ```suggestion
* Best Effort: We do not have continous integration tests for these, but we are fairly confident that gRPC C++ would work on them.
```

We will make our best effort to support them, and we welcome patches for such platforms, but we might need to declare bankruptcy on some issues.

| Operating System | Architectures | Versions | Support Level |
|------------------|---------------|----------|---------------|
| Linux            | x86, x64      | clang 3.4+, GCC 4.9+ | Supported |
| Windows 7+       | x86, x64      | Visual Studio 2015+  | Supported |

> | MacOS            | x86, x64      | clang 3.4+, XCode 7.2+ | Supported |

Reviewer: It'd be better to remove clang 3.4+ here since it's not common to use directly clang on MacOS. Rather, they commonly install XCode and use it.

In the C++ world, there's no universally accepted standard for managing project dependencies.
Therefore, gRPC supports several major build systems, which should satisfy most users.

## Supported Platforms

* Supported: These platforms are officially supported. We test our code on these
platform and have automated continuous integration tests for them.

* Best Effort: We do not have continous integration tests for these, but we are fairly confident that gRPC C++ would work on them.
We will make our best effort to support them, and we welcome patches for such platforms, but we might need to declare bankruptcy on some issues.

| Operating System | Architectures | Versions | Support Level |
|------------------|---------------|----------|---------------|

> | Linux            | x86, x64      | clang 3.4+, GCC 4.9+ | Supported |

Reviewer 3: Linux distribution does matter. E.g. we have a bunch of open issues around Alpine Linux support and we're being unclear about whether that's supported or not (Alpine is pretty widely used in Kubernetes deployments, but it uses a different libc library).

I think we fully support Debian, Ubuntu and CentOS / RedHat (for all of them we support the stable release + newer releases). For Alpine our support is "best effort".

> | Windows 7+       | x86, x64      | Visual Studio 2015+  | Supported |

Reviewer 3: Windows 7 is actually end of life since January 2020, so we might as well drop its official support from the list. Windows 10+ is enough.
